plasmapy/examples/plot_magnetic_statics.py

```diff
@@ -0,0 +1,104 @@
+# coding: utf-8
+"""
+Magnetostatic Fields
+================
+
+An example of using PlasmaPy's `Magnetostatic` class in `physics` subpackage.
+"""
+
+import plasmapy as pp
+from plasmapy import Plasma
+import numpy as np
+import astropy.units as u
+import matplotlib.pyplot as plt
+import itertools
+
+############################################################
+# Some common magnetostatic fields can be generated and added to a plasma object.
+
+# ## A dipole
+
+dipole = pp.physics.MagneticDipole(np.array([0, 0, 1])*u.A*u.m*u.m, np.array([0, 0, 0])*u.m)
+dipole
+
+############################################################
+# initialize a a plasma, where the magnetic field will be calculated on
+
+plasma = Plasma(domain_x=np.linspace(-2, 2, 30) * u.m,
+                domain_y=np.linspace(0, 0, 1) * u.m,
+                domain_z=np.linspace(-2, 2, 20) * u.m)
+
+############################################################
+# add the dipole field to it
+plasma.add_magneticstatic(dipole)
+
+X, Z = plasma.grid[0, :, 0, :], plasma.grid[2, :, 0, :]
+U = plasma.magnetic_field[0, :, 0, :].value.T  # because grid uses 'ij' indexing
+W = plasma.magnetic_field[2, :, 0, :].value.T  # because grid uses 'ij' indexing
+
+
+############################################################
+plt.figure()
+plt.axis('equal')
+plt.title('Dipole field in x-z plane, generated by a dipole pointing in the z direction')
+plt.streamplot(plasma.x.value, plasma.z.value, U, W)
```

Comment:
For added prettification:
* `plt.xlim(-2, 2)`
* I think you can just plot using Astropy quantities and there's something that lets you plot in quantity-enabled mode... I'll find that later.

Reply:
`plt.xlim(-2, 2)` does not work in my notebook. The plot is somehow forced to a fixed size.
But I find another way to get rid of the white margins.